From 16d090ee40979c9174975a0c358263e11b60096e Mon Sep 17 00:00:00 2001
From: jiansonx <jiansongx.xu@intel.com>
Date: Wed, 16 Jan 2019 21:34:54 +0800
Subject: [PATCH 1/3] MdePkg/UefiDebugLibStdErr: Avoid module not runtime safe.

BZ: https://bugzilla.tianocore.org/show_bug.cgi?id=1416
Remove DXE_RUNTIME_DRIVER and DXE_SMM_DRIVER type from the supported
module type list in the library INF file. Then, these library instances
don't support DXE_RUNTIME_DRIVER and DXE_SMM_DRIVER type any more.

Contributed-under: TianoCore Contribution Agreement 1.1
Signed-off-by: Jiansong Xu <jiansongx.xu@intel.com>
Cc: Gao, Liming <liming.gao@intel.com>
---
 MdePkg/Library/UefiDebugLibStdErr/UefiDebugLibStdErr.inf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/MdePkg/Library/UefiDebugLibStdErr/UefiDebugLibStdErr.inf b/MdePkg/Library/UefiDebugLibStdErr/UefiDebugLibStdErr.inf
index fe333b05ce..dd0a9e5d7e 100644
--- a/MdePkg/Library/UefiDebugLibStdErr/UefiDebugLibStdErr.inf
+++ b/MdePkg/Library/UefiDebugLibStdErr/UefiDebugLibStdErr.inf
@@ -22,7 +22,7 @@
   FILE_GUID                      = b57a1df6-ffdb-4247-a3df-3a562176751a
   MODULE_TYPE                    = UEFI_DRIVER
   VERSION_STRING                 = 1.0
-  LIBRARY_CLASS                  = DebugLib|DXE_CORE DXE_DRIVER DXE_RUNTIME_DRIVER DXE_SMM_DRIVER UEFI_APPLICATION UEFI_DRIVER
+  LIBRARY_CLASS                  = DebugLib|DXE_CORE DXE_DRIVER UEFI_APPLICATION UEFI_DRIVER
 
 
 #
-- 
2.16.2.windows.1

